subprojects/workers/src/main/java/org/gradle/workers/internal/WorkerDaemonClientsManager.java

```diff
@@ -137,4 +161,18 @@ public boolean isSatisfiedBy(WorkerDaemonClient client) {
             }
         }
     }
+
+    private class LogLevelChangeEventListener implements OutputEventListener {
+        @Override
+        public void onOutput(OutputEvent event) {
+            if (event instanceof LogLevelChangeEvent) {
+                LogLevelChangeEvent logLevelChangeEvent = (LogLevelChangeEvent) event;
+                if (currentLogLevel != logLevelChangeEvent.getNewLogLevel()) {
```

Comment:
This is unsafe, as you might see an old value of currentLogLevel outside the synchronized block. I also doubt the optimization is needed, given how infrequently this is called.